### PR TITLE
Update `swap_lon_axis()` to fix bounds for prime meridian cell

### LINKED ISSUE #790: [Bug]: lon_orient=(0, 360) tacking on an extra longitude
### What happened?

I am reading in a dataset with 360 longitudes that span -179.5 to 179.5 by 1 degree.  Instead I want the data to  run from 0.5 to 359.5 by 1 degree. Using ds = xc.open_dataset(filename, use_cftime = True, lon_orient=(0, 360)), it successfully reorients the longitudes to be positive definite (in degrees East), but it adds an additional longitude  such that I have both 359.5 and 360.

### What did you expect to happen? Are there are possible answers you came across?

I expected this to keep the same longitudes, but orient the data from 0.5 to 359.5 without adding that 360 degree point.

### Minimal Complete Verifiable Example (MVCE)

```python
On NERSC:

filename='/global/cfs/cdirs/m4581/obs4MIPs/obs4MIPs_input/MOHC/HadISST1-1/v20231114/HadISST.1.1.sst.nc'
ds0 = xc.open_mfdataset(filename, use_cftime = True)
ds1 = xc.open_mfdataset(filename, use_cftime = True, lon_orient=(0, 360))
ds0.longitude,ds1.longitude
```

### Relevant log output

```python
(<xarray.DataArray 'longitude' (longitude: 360)> Size: 1kB
 array([-179.5, -178.5, -177.5, ...,  177.5,  178.5,  179.5], dtype=float32)
 Coordinates:
   * longitude  (longitude) float32 1kB -179.5 -178.5 -177.5 ... 178.5 179.5
 Attributes:
     units:          degrees_east
     long_name:      Longitude
     standard_name:  longitude
     bounds:         longitude_bnds,
 <xarray.DataArray 'longitude' (longitude: 361)> Size: 1kB

 array([  0.5,   1.5,   2.5, ..., 358.5, 359.5, 360. ], dtype=float32)
 Coordinates:
   * longitude  (longitude) float32 1kB 0.5 1.5 2.5 3.5 ... 358.5 359.5 360.0
 Attributes:
     units:          degrees_east
     long_name:      Longitude
     standard_name:  longitude
     bounds:         longitude_bnds)
```

### Anything else we need to know?

_No response_

### Environment

xCDAT version 0.8.0

INSTALLED VERSIONS
------------------
commit: None
python: 3.13.2 | packaged by conda-forge | (main, Feb 17 2025, 14:10:22) [GCC 13.3.0]
python-bits: 64
OS: Linux
OS-release: 5.14.21-150500.55.97_13.0.78-cray_shasta_c
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: ('en_US', 'UTF-8')
libhdf5: 1.14.4
libnetcdf: 4.9.2

xarray: 2025.3.0
pandas: 2.2.3
numpy: 2.1.3
scipy: 1.15.2
netCDF4: 1.7.2
pydap: None
h5netcdf: None
h5py: None
zarr: None
cftime: 1.6.4
nc_time_axis: 1.4.1
iris: None
bottleneck: 1.4.2
dask: 2025.3.0
distributed: 2025.3.0
matplotlib: 3.10.1
cartopy: 0.24.1
seaborn: None
numbagg: None
fsspec: 2025.3.2
cupy: None
pint: 0.24.4
sparse: 0.15.4
flox: None
numpy_groupies: None
setuptools: 78.1.1
pip: 25.1
conda: None
pytest: None
mypy: None
IPython: 9.0.2
sphinx: None

### PR DESCRIPTION
**Pull request by tomvothecoder**
## Description

This pull request refactors and expands the longitude axis swapping tests, improves coverage for longitude bounds normalization (especially around the prime meridian), and moves the `_get_bounds_dim` helper and its tests to more appropriate locations. The changes also improve test clarity and maintainability by renaming and restructuring test cases.

 Closes #790 

**Longitude axis swapping and bounds normalization:**

* Added new tests to verify that swapping longitude axes is idempotent when the orientation does not change, and that longitude bounds are correctly normalized when crossing the prime meridian, both for single and multiple dimensions. This ensures correct behavior for edge cases and improves test coverage. [[1]](diffhunk://#diff-19c6a81ee8bff3232e316d71c8a4a0b08386bca61d11fb9da305a0cdee4721feR581-R655) [[2]](diffhunk://#diff-19c6a81ee8bff3232e316d71c8a4a0b08386bca61d11fb9da305a0cdee4721feL683-R848) [[3]](diffhunk://#diff-19c6a81ee8bff3232e316d71c8a4a0b08386bca61d11fb9da305a0cdee4721feL765-R872) [[4]](diffhunk://#diff-19c6a81ee8bff3232e316d71c8a4a0b08386bca61d11fb9da305a0cdee4721feL802-R898) [[5]](diffhunk://#diff-19c6a81ee8bff3232e316d71c8a4a0b08386bca61d11fb9da305a0cdee4721feL816-R912) [[6]](diffhunk://#diff-19c6a81ee8bff3232e316d71c8a4a0b08386bca61d11fb9da305a0cdee4721feL833-R958) [[7]](diffhunk://#diff-19c6a81ee8bff3232e316d71c8a4a0b08386bca61d11fb9da305a0cdee4721feL876-R973) [[8]](diffhunk://#diff-26199baaf391361625bd6773b9df4ed2677df4680a9650d2731510adec4898eaL1617-R1645)
* Refactored and renamed several test methods for clarity, and removed a test that raised errors for multiple prime meridian cells, as this case is now handled by normalization logic. [[1]](diffhunk://#diff-19c6a81ee8bff3232e316d71c8a4a0b08386bca61d11fb9da305a0cdee4721feL513-L551) [[2]](diffhunk://#diff-19c6a81ee8bff3232e316d71c8a4a0b08386bca61d11fb9da305a0cdee4721feL683-R848)

**Bounds dimension helper refactor:**

* Moved the `_get_bounds_dim` helper function import from `xcdat.bounds` to `xcdat.axis`, and relocated its associated tests from `tests/test_bounds.py` to `tests/test_axis.py`, aligning them with their usage and improving code organization. [[1]](diffhunk://#diff-19c6a81ee8bff3232e316d71c8a4a0b08386bca61d11fb9da305a0cdee4721feR8) [[2]](diffhunk://#diff-ab2e2d39d73a3689321c9a95d247ebb6147259bc14b3d022e780249baeecbfe5L15-R15) [[3]](diffhunk://#diff-ab2e2d39d73a3689321c9a95d247ebb6147259bc14b3d022e780249baeecbfe5L1062-L1097) [[4]](diffhunk://#diff-19c6a81ee8bff3232e316d71c8a4a0b08386bca61d11fb9da305a0cdee4721feL890-R1018)

These changes enhance the reliability of longitude axis operations and clarify the codebase by grouping related functionality and tests.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

If applicable:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass with my changes (locally and CI/CD build)
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have noted that this is a breaking change for a major release (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: xcdat/axis.py [hunk 1]
```diff
@@ -370,18 +370,23 @@ def _swap_lon_bounds(ds: xr.Dataset, key: str, to: tuple[float, float]):
...
-            p_meridian_index = _get_prime_meridian_index(ds[key])
+            p_meridian_index = _get_prime_meridian_index(ds_new[key])
 
             if p_meridian_index is not None:
-                ds = _align_lon_to_360(ds, ds[key], p_meridian_index)
+                ds_new = _adjust_bounds_for_prime_meridian(
+                    ds_new, key, p_meridian_index
+                )
+
+        return ds_new
```
Comment (tomvothecoder) [inline xcdat/axis.py:389]: Update to `_swap_lon_bounds()` here

File: xcdat/axis.py [hunk 2]
```diff
@@ -470,6 +475,58 @@ def _get_prime_meridian_index(lon_bounds: xr.DataArray) -> np.ndarray | None:
...
+    midpoint = ds_new[dim][p_meridian_index].item()
+    new_lower = midpoint - bounds_delta
+    new_upper = midpoint + bounds_delta
+
+    ds_new[key][p_meridian_index, :] = [new_lower, new_upper]
+
+    # Remove the extra cell to restore the original shape.
+    ds_new = ds_new.isel({dim: slice(0, -1)})
+
+    return ds_new
```
Comment (tomvothecoder) [inline xcdat/axis.py]: @pochedls The main code update of interest.

File: xcdat/axis.py [hunk 3]
```diff
@@ -470,6 +475,58 @@ def _get_prime_meridian_index(lon_bounds: xr.DataArray) -> np.ndarray | None:
...
+    midpoint = ds_new[dim][p_meridian_index].item()
+    new_lower = midpoint - bounds_delta
+    new_upper = new_lower + bounds_delta
+
+    ds_new[key][p_meridian_index, :] = [new_lower, new_upper]
+
+    # Remove the extra cell to restore the original shape.
+    ds_new = ds_new.isel({dim: slice(0, -1)})
+
+    return ds_new
```
Comment (tomvothecoder) [inline xcdat/axis.py:527]: The main function to analyze.

**Review by tomvothecoder — COMMENTED** [review]
~~Hey @pochedls, this PR is ready for review. It removes the extra longitude point generated when handling prime meridian cells and rectifies the bounds for the prime meridian cell (after swapping from (-180, 180) to (0, 360)). I still need to fix the unit tests.~~ Not ready yet (https://github.com/xCDAT/xcdat/pull/791#issuecomment-3282353529).

File: xcdat/axis.py (hunk 3 above)
Comment (tomvothecoder) [reply xcdat/axis.py]: I just made a fix for `    bounds_delta = abs(bounds.diff(bounds_dim).item()) / 2`.

**Review by tomvothecoder — COMMENTED** [review]
Hey @pochedls, this PR is ready for review. Refer to the [PR description](https://github.com/xCDAT/xcdat/pull/791#issue-3404119449) for more detailed info, if needed.

The main change to focus on is in `axis.py` under `def _normalize_wrap_bounds_0_to_360()`, which resolves the incorrect bounds for prime meridian cells. All other code to handle prime meridian cell code was removed (e.g,. cell splitting which adds additional cell). Besides that, refactoring was performed to make the code cleaner.

Unit tests were updated to pass and  the minimum example from #790 is now fixed.

### Output
Notice that last bounds has been fixed to convert [359, 0] -> [359, 360].
```python
Longitude Values (Last 5):
[355.5 356.5 357.5 358.5 359.5]


Longitude Bounds (Last 5):
[[355. 356.]
 [356. 357.]
 [357. 358.]
 [358. 359.]
 [359. 360.]]
```

### Script
```python
import xcdat as xc

filename = "/global/cfs/cdirs/m4581/obs4MIPs/obs4MIPs_input/MOHC/HadISST1-1/v20231114/HadISST.1.1.sst.nc"

ds = xc.open_dataset(filename)
ds_swapped = xc.open_dataset(filename, lon_orient=(0, 360))

# Print longitude values
print("Longitude Values:")
print(ds_swapped.longitude)
print("\n")

# Print longitude bounds
print("Longitude Bounds:")
print(ds_swapped.longitude_bnds)
```
